Sources/Rendering/OpenGL/PolyDataMapper/index.js

```diff
@@ -1501,38 +1510,50 @@ function vtkOpenGLPolyDataMapper(publicAPI, model) {
     const camm = model.openGLCamera.getKeyMatrixTime().getMTime();
     const progm = program.getLastCameraMTime();
 
-    if (progm !== camm) {
-      if (actor.getIsIdentity()) {
-        program.setUniformMatrix('MCDCMatrix', keyMats.wcdc);
-        if (program.isUniformUsed('MCVCMatrix')) {
-          program.setUniformMatrix('MCVCMatrix', keyMats.wcvc);
-        }
-        if (program.isUniformUsed('normalMatrix')) {
-          program.setUniformMatrix3x3('normalMatrix', keyMats.normalMatrix);
-        }
-      }
-      if (program.isUniformUsed('cameraParallel')) {
-        program.setUniformi('cameraParallel', cam.getParallelProjection());
+    const shiftScaleEnabled = cellBO.getCABO().getCoordShiftAndScaleEnabled();
+    const inverseShiftScaleMatrix = shiftScaleEnabled
+      ? cellBO.getCABO().getInverseShiftAndScaleMatrix()
+      : null;
+
+    const actorIsIdentity = actor.getIsIdentity();
+    const actMats = actorIsIdentity ? null : model.openGLActor.getKeyMatrices();
+
+    if (progm !== camm || !actorIsIdentity) {
```

Comment:
Do I understand correctly that if camera has not been changed and if actor matrice was not identity but just became identity, then no uniform are recomputed ? Is it really what we want ?

Reply:
Before this change, uniforms were computed in two cases:
* If the camera has been changed and the actor matrix is identity
* If the actor matrix is not identity

This change keeps the same logic as before (updates uniforms if the camera has been changed or the actor matrix is not identity).

In your example, the uniforms would indeed not be recomputed, as would be the case in the previous version. Maybe a fix would be to also update uniforms if the actor's mtime is different to the program's.

Reply:
After discussion, this should be a separate issue.